cronjob_scripts/trigger-package-build.py

```diff
@@ -0,0 +1,184 @@
+    # to avoid getting stuck on unbuildable crates.
+    #
+    # For now I'm trying to be roughly backwards compatible with the old script's numbers,
+    # but I think that shuffling first is an important bugfixt. Once we are using crates.io sparse
+    # index api without rate-limits, we can think about checking more crates.
+    crates_to_check = random.sample(list(possible_crates), 4 * check_limit)
+    if os.environ.get("RECHECK"):
+        crates_to_check = ["cargo-quickinstall"] + crates_to_check
+
+    repo_url = get_repo_url()
+    branch = get_branch()
+    builds_scheduled = 0
+    for crate in crates_to_check:
+        print(f"Checking {crate} for {target_arch}")
+        version = get_current_version_if_unbuilt(
+            repo_url=repo_url, crate=crate, target_arch=target_arch
+        )
+        if not version:
+            continue
+
+        no_default_features = "false"
+        if crate == "gitoxide":
+            features = "max-pure"
+            no_default_features = "true"
+        elif crate == "sccache":
+            features = "vendored-openssl"
+        else:
+            features = ",".join(
+                feat for feat in version["features"].keys() if "vendored" in feat
+            )
+
+        workflow_run_input = {
+            "crate": crate,
+            "version": version["vers"],
+            "target_arch": target_arch,
+            "features": features,
+            "no_default_features": no_default_features,
+            "build_os": build_os,
+            "branch": branch,
+        }
+        print(f"Attempting to build {crate} {version['vers']} for {target_arch}")
+        subprocess.run(
+            ["gh", "workflow", "run", "build-package.yml", "--json", f"--ref={branch}"],
+            input=json.dumps(workflow_run_input).encode(),
+            check=True,
+        )
+        builds_scheduled += 1
+        if builds_scheduled >= check_limit:
+            print("Scheduled enough builds, exiting")
+            break
+        time.sleep(30)
+
+
+def get_popular_crates():
+    """
+    List of crates that are popular and should be checked in addition to the requested crates.
+
+    In the future, this list might be generated from the crates.io db dump (see
+    https://github.com/cargo-bins/cargo-quickinstall/issues/268#issuecomment-2329308074),
+    but probably not every time we run this script. We will need to think about this.
+    """
+    with open(
+        Path(__file__).resolve().parent.parent.joinpath("popular-crates.txt"), "r"
+    ) as file:
+        return [
+            line.strip()
+            for line in file
+            if line.strip() != "" and not line.startswith("#")
+        ]
+
+
+def get_excludes(tracking_worktree_path: str, days: int, max_failures: int):
+    """
+    if a crate has reached `max_failures` failures in the last `days` days then we exclude it
+    """
+    failures = Counter()
+
+    failure_filenames = list(Path(tracking_worktree_path).glob("*-*-*"))
```

Comment:
For these files, I think we don't have any mechanism to delete them?

So it'd keep growing, I'd recon we need to add a cleanup mechanism in another PR, otherwise this would get inefficient eventually.

Reply:
backlogged as https://github.com/cargo-bins/cargo-quickinstall/issues/276